auslib/test/web/test_client.py

```diff
@@ -910,6 +911,23 @@ def testNonSubstitutedUrlVariablesReturn404(self):
                 self.assertEqual(ret.status_code, 404)
                 self.assertFalse(mock_cr_view.called)
 
+    @given(just(1))
+    def testUpdateQueryVersion1ToEnsureOsVersion(self, v1_qv):
+        query_version = extract_query_version('/update/{}/a/b/c/update.xml'.format(v1_qv))
+        self.assertEqual(query_version, v1_qv)
+
+    @given(just("x1"), just("1x"))
+    def testUpdateInvalidQueryVersion(self, invalid_prefix, invalid_suffix):
+        query_version = extract_query_version('/update/x1/a/b/c/update.xml')
+        self.assertEqual(query_version, 0)
+        query_version = extract_query_version('/update/1x/a/b/c/update.xml')
+        self.assertEqual(query_version, 0)
+
+    @given(integers(min_value=1, max_value=100))
```

Comment:
I _think_ you can combine this with the first test by using: `@example`, like https://hypothesis.readthedocs.io/en/latest/details.html#providing-explicit-examples. r=me with that changed.